examples/framecount.py

```diff
@@ -6,112 +6,37 @@
-
 import logging
-# logging is primarily to set the logging level. Can also be inserted into
-# the sample model and view.
-# time is used by the view to print the unix time stamp.
-
-import ppb.engine as engine
-from ppb.event import Tick, Quit
-from ppb.utilities import Publisher
-# import ppb.engine is the primary import.
-# ppb.event to grab any events you plan to listen to.
-# ppb.utilities has multiple base classes used throughout the library.
-# in this case we'll be using a Publisher as a scene.
-
-
-logging.basicConfig(level=logging.INFO)
-# Set the logging level
+import time
+import ppb
 
 
-class TestController(object):
-    """
-    A controller that raises a Quit event after the given time.
-    """
-
-    def __init__(self, scene, length):
-        """
-        Pass in the scene and the number of seconds to run the
-        script.
-
-        :param scene: Publisher
-        :param length: number
-        :return: TestController
-        """
-        self.countdown = length
-        # need to keep track of how much time is left.
-        # Alternatively could keep the time to run and accumulate.
-
-        scene.subscribe(Tick, self.tick)
-        # subscribe to Tick events with the scene.
-
-    def tick(self, event):
-        """
-        Advance the countdown and raise a Quit event when complete.
-
-        :param event: Tick
-        :return:
-        """
-        self.countdown += -1 * event.sec
-        # Advance countdown. Use -1 * time to be compatible with Python2
-
-        if self.countdown <= 0:
-            engine.message(Quit())
-            # Raise the quit event.
-
-
-class TestView(object):
+class TestScene(ppb.BaseScene):
     """
     A view that prints a test line in place of rendering frames.
     """
 
-    def __init__(self, scene, target_fps):
-        """
-
-        :param scene: Publisher
-        :param target_fps: number of frames per second to render
-        :return:
-        """
-        self.target_render_time = 1/target_fps
-        # Get amount of time between render calls.
+    duration = 1.0
 
-        self.countdown = self.target_render_time
-        # A countdown to control render speed.
-        self.count = 0
-        # A simple accumulator to count frames.
-        scene.subscribe(Tick, self.tick)
-        # Subscribe to the scene.
+    def __init__(self, *p, **kw):
+        super().__init__(*p, **kw)
 
-    def tick(self, event):
-        """
-        Advance render countdown, render, and reset counter.
+        self.frames = 0
+        self.start_time = time.monotonic()
 
-        :param event: Tick
-        :return:
-        """
-        self.countdown += -1 * event.sec
-        # Advance countdown. Use -1 * time to be compatible with Python2
+    def on_update(self, event, signal):
+        # Fires at the update rate (~60 times a second)
+        t = time.monotonic() - self.start_time
+        if t >= self.duration:
+            signal(ppb.events.Quit())
```

Comment:
Clever way to kill it.

Might be better to accumulate via Update.time_delta? (Only because it explains that feature in the sample.)

Reply:
No, because then we'd only update the time every {{tick rate}} instead of every {{frame rate}}.

Reply:
And I think `targets.py` adequately demonstrates Update.

Reply:
Um, confused at how that changes this bit of code? Using monotonic versus accumulating update time to decide when to kill the process doesn't change the print rate.

Reply:
Oh, I see.

You're not wrong, but there's also arguments to be made about consistency or something.